devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -1365,6 +1866,579 @@ def deleteCode(codeID) {
+	log.debug "In requestDailyRepeatingSchedule() with codeID := $codeID"
+	secure(zwave.scheduleEntryLockV3.scheduleEntryLockDailyRepeatingGet(scheduleSlotId: 1, userIdentifier: codeID))
+}
+
+/**
+ * Returns the command for week day schedule get
+ *
+ * @param codeID: The code slot number
+ *
+ * @param slotID: The schedule slot number
+ *
+ * @return The command for week day schedule get
+ */
+private def requestWeekDaySchedule(slotID, codeID) {
+	secure(zwave.scheduleEntryLockV3.scheduleEntryLockWeekDayGet(scheduleSlotId: slotID, userIdentifier: codeID))
+}
+
+/**
+ * Returns the command for week day schedule get for all schedule slot ids
+ *
+ * @param codeID: The code slot number
+ *
+ * @return The command for week day schedule get for all schedule slot ids
+ */
+private def requestAllWeekDaySchedule(codeID) {
+	def cmds = []
+	for (def i = 0; i <= 6; i++) {
+		cmds << requestWeekDaySchedule((i + 1), codeID)
+	}
+	cmds = delayBetween(cmds, 200)
+	cmds
+}
+
+/**
+ * Returns the command for year day schedule get
+ *
+ * @param codeID: The code slot number
+ *
+ * @return The command for year day schedule get
+ */
+private def requestYearDaySchedule(codeID) {
+	secure(zwave.scheduleEntryLockV3.scheduleEntryLockYearDayGet(scheduleSlotId: 1, userIdentifier: codeID))
+}
+
+/**
+ * Checks if a particular bit is set in a number
+ *
+ * @param weekDayBitmask: The week day bit mask
+ *
+ * @param bitNum: The bit number to check
+ *
+ * @return - true if the bit is set, false otherwise
+ */
+def isBitSet(weekDayBitmask, bitNum) {
+	def mask = 1 << bitNum
+	if ((weekDayBitmask & mask) != 0) {
+		return true
+	}
+	return false
+}
+
+/**
+ * Calculates the duration in hours and minutes
+ *
+ * @param startHour: The start hour
+ *
+ * @param startMin: The start minute
+ *
+ * @param endHour: The end hour
+ *
+ * @param endMin: The end minute
+ *
+ * @return The map with duration in hours and minutes
+ */
+private Map getDurationInHoursAndMinutes(startHour, startMin, endHour, endMin) {
+	def startDate = new Date()
+	startDate.set(hourOfDay: startHour, minute: startMin, second: 0, year: 2000, month: 1, date: 1)
+
+	def endDate = new Date()
+	endDate.set(hourOfDay: endHour, minute: endMin, second: 0, year: 2000, month: 1, date: 1)
+
+	long difference = endDate.getTime() - startDate.getTime();
+	difference = difference / 1000
+	def durSeconds = difference % 60
+	difference = (difference - durSeconds) / 60
+	def durMin = difference % 60
+	difference = (difference - durMin) / 60
+	def durHour = difference % 24
+	return [durHour: durHour, durMin: durMin]
+}
+
+/**
+ * Removes the specified code if from lockCodesData and updates the attribute
+ *
+ * @param codeID: The code slot number
+ */
+def clearUserInfo(codeID) {
+	log.trace "[DTH] Executing 'clearUserInfo()' by ${device.displayName} with codeID := $codeID"
+	def lockCodesData = parseJson(device.currentValue("lockCodesData") ?: "{}") ?: [:]
+	codeID = codeID.toString()
```

Comment:
as a general style comment, very simple statements like this don't need to be broken out into one line